src/external/dftk_to_wannier90.jl

```diff
@@ -0,0 +1,605 @@
+            for coord in atoms[i][2]              #loop over atoms of the same element
+                write(f,symbol*"  "*"$(coord[1])  $(coord[2])  $(coord[3])"*"\n")  
+            end
+        end
+        write(f,"end atoms_frac"*"\n"^2)
+
+        ## Projection bloc : message for the user.
+        write(f,"""
+        ! Add a projection bloc only with the option : guess = "win" in dftk2wan_wannierization_files.
+        ! Otherwise, DFTK uses by default gaussian guesses on specified centers 
+        ! or the SCDM method if guess = "SCDM" (not yet implemented).
+
+
+        """)
+
+
+        ## k points
+        write(f,"!"^20*" k_points"*"\n"^2)
+
+        # Bands_plot bloc
+        if bands_plot
+            trash,labels,path = high_symmetry_kpath(basis.model)
+            write(f,"begin kpoint_path"*"\n")
+            for i in 1:size(path[1],1)-1
+                ## write path section A -> B
+                A,B = path[1][i],path[1][i+1]
+                # kpoints are writen in the file with normalized coordinates
+                k_A = round.(DFTK.normalize_kpoint_coordinate.(get!(labels,A,1)), digits = 5)
+                k_B = round.(DFTK.normalize_kpoint_coordinate.(get!(labels,B,1)), digits = 5)
+                # Wannier90 convention for the Γ point is G
+                if A == "\\Gamma"
+                    A = "G"
+                end
+                if B == "\\Gamma"
+                    B = "G"
+                end
+                write(f,"$A $(k_A[1]) $(k_A[2]) $(k_A[3]) $B $(k_B[1]) $(k_B[2]) $(k_B[3])"*"\n") 
+            end
+            write(f,"end kpoint_path"*"\n")
+            write(f,"bands_plot = T"*"\n"^2)
+        end
+        
+        # Mp grid
+        if kgrid[1]*kgrid[2]*kgrid[3] !== size(scfres.ψ,1)
+            error("Given kgrid doesn't match the kgrid used for scf calculation")
+        end
+        write(f,"mp_grid : $(kgrid[1]) $(kgrid[2]) $(kgrid[3])"*"\n")
+        if kgrid == [1,1,1]
+            write(f,"gamma_only      = true"*"\n")
+        else
+            write(f,"\n")
+        end
+        
+        # kpoints bloc
+        write(f,"begin kpoints"*"\n")
+        for i in 1:size(scfres.ψ,1)
+            for x in basis.kpoints[i].coordinate
+                write(f,"$x ")
+            end
+            write(f,"\n")
+        end
+        write(f,"end kpoints"*"\n"^2)
+
+        @info "File generated.",f
+    end
+    
+    
+end
+
+
+"""
+Read the .nnkp file provided by the preprocessing routine of Wannier90 (i.e. "wannier90.x -pp prefix")
+Returns: 
+1) the array 'nn_kpts' of k points, they respective neirest neighbours and associated shifing vectors (non zero if the neighbour of is located in another cell).
+2) the number 'nn_num' of neighbours per k point. 
+2) the array 'projs' of projections. Each line contains the informations for one projection :  [center],[quantum numbers: l,mr,r],[ref. z_axis],[ref. x_axis],α = Z/a
+"""
+function read_nnkp_file(prefix::String)
```

Comment:
I don't really know the format of the nnkp files, but this parser code looks a little complicated at some parts and likely can be made a little easier to read, but without understanding the file format there is little I can say. We should talk about it when we get closer to merge this.